src/storage/core.rs

```diff
@@ -710,13 +721,37 @@ where
                             .with_context(|| {
                                 anyhow!(format!("failed to save corrupted blob {:?}", file))
                             })?;
+                        corrupted += 1;
                     } else {
                         return Err(e.context(msg));
                     }
                 }
             }
         }
-        Ok(blobs)
+        Ok((blobs, corrupted))
+    }
+
+    async fn count_old_corrupted_blobs(files: &[DirEntry], config: &Config) -> usize {
+        let dir_content = files.iter().map(DirEntry::path);
+        let dirs = dir_content.filter(|path| 
+            path.is_dir() && path.ends_with(config.corrupted_dir_name()));
```

Comment:
Use `file_name` instead of `ends_with`. `ends_with` will not work, for example, if the path ends with `/`